lib/src/connection.dart

```diff
@@ -451,34 +456,41 @@ abstract class _PostgreSQLExecutionContextMixin
       columnDescriptions = await _connection._oidCache
           ._resolveTableNames(this, columnDescriptions);
     }
-    final metaData = _PostgreSQLResultMetaData(columnDescriptions);
-
-    return _PostgreSQLResult(
-        queryResult.affectedRowCount,
-        metaData,
-        queryResult.value
-            .map((columns) => _PostgreSQLResultRow(metaData, columns))
-            .toList());
+    final metaData = _PostgreSQLResultMetaData(columnDescriptions!);
+
+    return queryResult != null
+        ? _PostgreSQLResult(
+            queryResult.affectedRowCount,
+            metaData,
+            queryResult.value!
+                .map((columns) => _PostgreSQLResultRow(metaData, columns))
+                .toList())
+        : null;
   }
 
   @override
-  Future<List<Map<String, Map<String, dynamic>>>> mappedResultsQuery(
+  Future<List<Map<String?, Map<String, dynamic>>>> mappedResultsQuery(
       String fmtString,
-      {Map<String, dynamic> substitutionValues,
-      bool allowReuse,
-      int timeoutInSeconds}) async {
+      {Map<String, dynamic> substitutionValues = const {},
+      bool allowReuse = false,
+      int timeoutInSeconds = 30}) async {
     final rs = await query(
       fmtString,
       substitutionValues: substitutionValues,
       allowReuse: allowReuse,
       timeoutInSeconds: timeoutInSeconds,
     );
+
+    if (rs == null) {
+      return Future.value();
+    }
     return rs.map((row) => row.toTableColumnMap()).toList();
   }
 
   @override
-  Future<int> execute(String fmtString,
-      {Map<String, dynamic> substitutionValues, int timeoutInSeconds}) async {
+  Future<int?> execute(String fmtString,
+      {Map<String, dynamic> substitutionValues = const {},
```

Comment:
Same here: I'd rather keep these optional parameters as nullable and set a default value just before we use them. Especially because it is important to be consistent about it, and there `timeoutInSeconds` is nullable.